libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -1463,6 +1467,14 @@ void ExpressionCompiler::appendArithmeticOperatorCode(Token::Value _operator, Ty
 		break;
 	case Token::Mul:
 		m_context << Instruction::MUL;
+		if (fixedType)
+		{
+			// TODO: not really tested
+			solUnimplementedAssert(fixedFactor < (u256(1) << (fixedType->numBits() / 2)), "Not yet implemented - large FixedPointType multiplication");
+			m_context << fixedFactor
+					  << Instruction::SWAP1
+					  << (isSigned ? Instruction::SDIV : Instruction::DIV);
```

Comment:
If you use sdiv, you have to ensure that the operands are properly cleaned before you use it. Is that the case?

Reply:
I guess yse, because the sdiv has been here before - just checking. Also a test for exactly that would be nice :)

Reply:
No, that is not guaranteed. The only way I could think of to ensure this would be to do a proper conversion through CompilerUtils. This probably also happens around here with multiplication.

Reply:
Okay, will do